Send all errors to STDERR

scjody opened the pull request:
This makes all errors go to STDOUT consistently, and future-proofs us if Dokku adds fancier error handling later.

@bpostlethwaite Please review.
@josegonzalez Feel free to take a look if you like.

josegonzalez commented on pre-build:
```diff
@@ -27,7 +28,7 @@ fi
   if [[ -n "$DOKKU_SUPER_USER" ]] && [[ "$NAME" != "$DOKKU_SUPER_USER" ]]; then
-    echo "Only $DOKKU_SUPER_USER can modify a repository if the ACL is empty" >&2;
+    dokku_log_fail "Only $DOKKU_SUPER_USER can modify a repository if the ACL is empty"
     exit 1;
```
Don't need this.

josegonzalez commented on pre-build:
```diff
@@ -37,6 +38,6 @@ fi
 if [[ ! -f "$ACL_FILE" ]] && [[ "$NAME" != "$DOKKU_SUPER_USER" ]]; then
-  echo "User $NAME does not have permissions to modify this repository" >&2
+  dokku_log_fail "User $NAME does not have permissions to modify this repository"
   exit 2;
```
This won't exit 2 anymore.

josegonzalez commented on user-auth:
```diff
@@ -27,23 +27,23 @@ done
 
-echo "User $SSH_NAME does not have permissions to run $CMD"
+dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD"
 exit 1
```
Don't need this.

josegonzalez commented on user-auth:
```diff
@@ -27,23 +27,23 @@ done
 
-    echo "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist" >&2
+    dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist"
     exit 1
```
Don't need this.

josegonzalez commented on user-auth:
```diff
@@ -27,23 +27,23 @@ done
     if ! ( verify_app_name "$APP" 2>/dev/null ); then
-      echo "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist" >&2
+      dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist"
       exit 1
```
Don't need this.

josegonzalez commented on user-auth:
```diff
@@ -27,23 +27,23 @@ done
     if [[ -z "$2" ]]; then
-      echo "An app name is required"
+      dokku_log_fail "An app name is required"
       exit 1
```
Don't need this.

josegonzalez commented on pre-build:
```diff
@@ -14,10 +15,10 @@ if [[ -z "$NAME" ]] ; then
+      "safety.  Please check the \"dokku-acl\" documentation for how" \
+      "to enable command line usage."
 
     exit 1
```
Don't need this.

josegonzalez commented on commands:
```diff
@@ -66,7 +69,7 @@ case "${1:-}" in
     then
-      echo "Please specify a user name" >&2
+      dokku_log_fail "Please specify a user name"
       exit 1;
```
Don't need this.

josegonzalez commented on commands:
```diff
@@ -55,7 +58,7 @@ case "${1:-}" in
     then
-       echo "User already has permissions to push to this repository"
+       dokku_log_fail "User already has permissions to push to this repository"
        exit 2;
```
This won't be called.

josegonzalez commented on commands:
```diff
@@ -46,7 +49,7 @@ case "${1:-}" in
     then
-      echo "Please specify a user name" >&2
+      dokku_log_fail "Please specify a user name"
       exit 1;
```
Don't need this.

josegonzalez commented on commands:
```diff
@@ -12,21 +15,21 @@ ACL_PATH="$DOKKU_ROOT/$APP/acl"
   then
-    echo "You can only modify ACL using local dokku command on target host" >&2
+    dokku_log_fail "You can only modify ACL using local dokku command on target host"
     echo 3;
```
This and the exit won't be called.

josegonzalez commented on commands:
```diff
@@ -12,21 +15,21 @@ ACL_PATH="$DOKKU_ROOT/$APP/acl"
   then
-    echo "No such app: $1" >&2
+    dokku_log_fail "No such app: $1"
     echo 2;
```
This and the exit won't be called.

josegonzalez commented on commands:
```diff
@@ -12,21 +15,21 @@ ACL_PATH="$DOKKU_ROOT/$APP/acl"
   then
-    echo "You must provide an app name!" >&2
+    dokku_log_fail "You must provide an app name!"
     echo 1;
```
This and the exit won't be called.

scjody commented:
@josegonzalez Sorry, I missed the `exit 1` in `dokku_log_fail`. I fixed the issues and also refactored `check_app` since it was a bit strange. How does this look now?

@bpostlethwaite OK with you?

bpostlethwaite commented:
LGTM :dancer:
